src/node_api.cc

```diff
@@ -435,6 +428,78 @@ class ThreadSafeFunction : public node::AsyncResource {
   bool handles_closing;
 };
 
+class AsyncContext {
+ public:
+  AsyncContext(node_napi_env env,
+               v8::Local<v8::Object> resource_object,
+               const v8::Local<v8::String> resource_name)
+      : env_(env) {
+    async_id_ = node_env()->new_async_id();
+    trigger_async_id_ = node_env()->get_default_trigger_async_id();
+    resource_.Reset(node_env()->isolate(), resource_object);
+    resource_.SetWeak(this,
```

Comment:
Is it needed to pass a callback to `setWeak()` which only calls `resource_.Reset()`? To my understanding `setWeak()` without parameter does this automatically.

Reply:
See https://github.com/nodejs/node/blob/master/deps/v8/include/v8.h#L443 , it is required by v8 that the first (and the only one here) embedder callback should reset the global handle.

Reply:
ok, thanks. I have seen it without callback e.g. [there ](https://github.com/nodejs/node/blob/bfc906906f8fdbed93681ac7079644f859d12e11/src/node_util.cc#L204) therefore I assumed it is possible.